### PR TITLE
test: add comprehensive unit tests for uncovered modules

### PR DESCRIPTION
**Pull request by b3y0urs3lf**
Add 324 new tests across 12 test files covering previously untested areas:

- NostrClient (72 tests): connection lifecycle, message handling, publishing, subscriptions, reconnection, NIP-17 messaging, nametag queries, disconnect cleanup, configuration combinations, concurrent operations
- Schnorr edge cases (33 tests): input validation for wrong key/message lengths, tampered signatures, cross-key verification
- EventKinds comprehensive (49 tests): isReplaceable, isEphemeral, isParameterizedReplaceable boundary testing, getName for all kinds
- Nametag edge cases (42 tests): phone heuristic edge cases, normalization, hash determinism, display formatting, country code handling
- Event mutability (25 tests): tags/content mutability, JSON serialization, ID calculation determinism, isValidEventData edge cases
- NIP-44 edge cases (29 tests): corrupted data, version byte validation, payload boundaries, padding algorithm verification
- KeyManager NIP-44 (18 tests): encrypt/decrypt with bytes/hex keys, conversation key symmetry
- Event edge cases (15 tests): empty/many/unicode tags, long content, deterministic IDs, JSON roundtrip
- Security critical paths (15 tests): key clearing, memory zeroing, copy semantics, NIP-17 sender anonymity, timestamp randomization
- WebSocket adapter (11 tests): extractMessageData for various data types
- NIP-04 edge cases (9 tests): corrupted ciphertext, invalid IV, GZIP errors
- CallbackEventListener (6 tests): callback invocation patterns

Known issues documented in tests:
- Event.tags is mutable (modifying breaks signature)
- Event.toJSON() returns same tags reference (not a copy)
- BIP-340 signatures use randomness (not deterministic)

Total test count: 274 -> 598 tests (118% increase) All tests pass, no source code modifications.

### CODE REVIEW

File: tests/bdd-test-scenarios.feature [hunk 1]
```diff
@@ -0,0 +1,1206 @@
...
+      | input_len | expected |
+      | 1         | 32       |
+      | 31        | 32       |
+      | 32        | 32       |
+      | 33        | 64       |
+      | 37        | 40       |
+      | 63        | 64       |
+      | 64        | 64       |
+      | 65        | 80       |
+      | 100       | 112      |
```
[the remark this answers is not in the document]
Comment (b3y0urs3lf) [reply tests/bdd-test-scenarios.feature]: Good catch — the BDD table values were indeed wrong. However, your suggested fix is also partially incorrect for input 65.
                                                                                                                                                                                              
  Tracing the actual implementation:                                          

  const nextPow2 = 1 << Math.ceil(Math.log2(unpaddedLen));
  const chunk = Math.max(32, nextPow2 >> 3);
  return Math.ceil(unpaddedLen / chunk) * chunk;

  For input 65: nextPow2=128, chunk=max(32, 16)=32, ceil(65/32)*32 = 96 — not 128.

  Your suggestion of 65 → 128 treats nextPow2 as the final result, skipping the chunk-rounding step. The algorithm doesn't jump to the next power of 2 — it divides the range into 8 equal
  chunks and rounds up to the nearest chunk boundary.

  Correct values applied in our fix:

  ┌───────┬─────────────┬─────────────────┬────────┐
  │ input │ was (wrong) │ your suggestion │ actual │
  ├───────┼─────────────┼─────────────────┼────────┤
  │ 37    │ 40          │ 64              │ 64     │
  ├───────┼─────────────┼─────────────────┼────────┤
  │ 65    │ 80          │ 128             │ 96     │
  ├───────┼─────────────┼─────────────────┼────────┤
  │ 100   │ 112         │ 128             │ 128    │
  └───────┴─────────────┴─────────────────┴────────┘

**Review by MastaP — CHANGES_REQUESTED** [review]
1. BDD feature file is dead weight — bdd-test-scenarios.feature (1206 lines) isn't wired to any runner. It's documentation pretending to be a test artifact.
  Should either be connected to a framework (cucumber-js) or moved to docs.
  2. Polling loop in NostrClient test — nostr-client.test.ts line ~1025 uses a setTimeout polling loop (50 iterations x 20ms) to wait for async encryption in the
  publishEncryptedMessage test. Fragile; should use flushMicrotasks() consistently like the other tests do.
  3. Merge conflict risk — The PR branch is based on c11f996 (v0.3.2) and is missing ff9b11f (nametag validation) from main. The existing nametag.test.ts was
  modified in that commit, so merging will likely conflict. The PR needs a rebase onto current main.
  4. No tests for isValidNametag() — The nametag validation function added in ff9b11f has no edge-case coverage in the new nametag-edge-cases.test.ts. After
  rebasing, tests for that function should be added.
